yarn-project/noir-contracts/src/contracts/native_token_contract/src/main.nr

```diff
@@ -315,7 +317,28 @@ contract NativeToken {
     ) {
         let storage = Storage::init(Option::some(&mut context), Option::none());
 
-        // Remove user balance
+        // If `from != sender` then we use the is_valid function to check that the message is approved.
+        if (from != context.msg_sender()) {
+            // Compute the message hash, should follow eip-712 more here.
+            // @todo @lherskind, probably need a separate generator index and address of the 
+            // @todo @lherskind Currently this can be used multiple times since it is not nullified. 
+            //                  We can do a simple nullifier to handle that in here. Spends only 32 bytes onchain.
+            // @todo @LHerskind Is to be solved as part of https://github.com/AztecProtocol/aztec-packages/issues/1743
+            let message_field: Field = std::hash::pedersen_with_separator([
+                    0x90785014, 
+                    from, 
+                    to, 
+                    amount
+                ], 
+                GENERATOR_INDEX__SIGNATURE_PAYLOAD
```

Comment:
Im not sure I like that generator indexes are exposed directly to the programmer. I feel like there should be a helper function that takes in an array of inputs with a selector or something to abstract this from them. 

It feels like a certain footgun

Reply:
Ideally it should not be this generator at all. Think it is better with a separate generator such that payloads and other signatures don't sit in the same domain.